Sources/SafeDICore/Generators/ScopeGenerator.swift

```diff
@@ -45,27 +45,19 @@ actor ScopeGenerator {
         }
         self.property = property
         self.propertiesToGenerate = propertiesToGenerate
-        forwardedProperties = scopeData.forwardedProperties
-        propertiesMadeAvailableByChildren = Set(
```

Comment:
calculating this as a union recursive list created issues when we re-instantiate the same property multiple times in the tree. We now inspect each tree individually (depth first) to avoid this trap.